app/controllers/subject_rule_effects_controller.rb

```diff
@@ -2,39 +2,35 @@ class SubjectRuleEffectsController < ApplicationController
   responders :flash
 
   def index
-    authorize workflow, policy_class: SubjectRuleEffectPolicy
+    authorize workflow, :validate_workflow, policy_class: SubjectRuleEffectPolicy
     effects = policy_scope(SubjectRuleEffect).where(subject_rule_id: params[:subject_rule_id])
     respond_with effects
   end
 
   def show
-    # TODO: Add this to the SubjectRuleEffect policy class
-    # and pass in the found subject_rule_effect
-    authorize workflow
+    authorize workflow, :validate_workflow, policy_class: SubjectRuleEffectPolicy
     @subject_rule_effect = policy_scope(SubjectRuleEffect).find(params[:id])
     respond_with workflow, subject_rule, @subject_rule_effect
   end
 
   def new
-    authorize workflow, :edit?
+    authorize workflow, :validate_workflow, policy_class: SubjectRuleEffectPolicy
     @subject_rule_effect = SubjectRuleEffect.new(action: params[:action_type], subject_rule: subject_rule)
     respond_with workflow, subject_rule, @subject_rule_effect
   end
 
   def edit
-    # TODO: the finders for SubjectRuleEffect in this controller
-    # are ripe for a preload / eager_load as we'll use the
-    # subject_rule.workflow relation in the policy scopes
-    @subject_rule_effect = SubjectRuleEffect.find(params[:id])
+    @subject_rule_effect = SubjectRuleEffect.includes(subject_rule: [:workflow]).find(params[:id])
     authorize @subject_rule_effect
 
     respond_with workflow, subject_rule, @subject_rule_effect
   end
 
   def create
-    authorize workflow, policy_class: SubjectRuleEffectPolicy
+    authorize workflow, :validate_workflow, policy_class: SubjectRuleEffectPolicy
 
     @subject_rule_effect = SubjectRuleEffect.new(effect_params)
+    authorize @subject_rule_effect
```

Comment:
> (the policy create method would dig into the linked workflow of the subject rule effect so pretty much the same)

Actually, the policy create method right now only validates the linked project for the subject set or collection, specifically:
`record.config['subject_set']['links']['project_id']` (where the record is a subject rule effect). I do see what you're saying though. It would actually make sense to just put all of that validation in the same method, i.e. validate `record.subject_rule.workflow.project_id` in that same policy method.

Will update accordingly.